history/history-network.md

```diff
@@ -236,45 +236,79 @@ content          = SSZ.serialize(block_header_with_proof)
+find_content_ephemeral_headers_key = Container(block_hash: Bytes32, ancestor_count: uint8)
 selector                 = 0x04
 
 BlockHeader              = ByteList[MAX_HEADER_LENGTH]
 ephemeral_header_payload = List(BlockHeader, limit=MAX_EPHEMERAL_HEADER_PAYLOAD)
 
 content                  = SSZ.serialize(ephemeral_header_payload)
-content_key              = selector + SSZ.serialize(ephemeral_headers_key)
+content_key              = selector + SSZ.serialize(find_content_ephemeral_headers_key)
 ```
 
-The `ephemeral_headers_key` encodes a request for headers anchored to the block
-hash indicated by `ephemeral_headers_key.block_hash`.  The
-`ephemeral_headers_key.ancestor_count` **MUST** be in the inclusive range
+The `find_content_ephemeral_headers_key` encodes a request for headers anchored to the block
+hash indicated by `find_content_ephemeral_headers_key.block_hash`. The
+`find_content_ephemeral_headers_key.ancestor_count` **MUST** be in the inclusive range
 0-255.
 
 The `ephemeral_header_payload` is an SSZ list of RLP encoded block header
-objects.  This object is subject to the following validity conditions.
+objects. This object is subject to the following validity conditions.
 
+* This content type **MUST** only be used for `FindContent` requests
 * The list **MAY** be empty which signals that the responding node was unable to fulfill the request.
-* The first element in the list **MUST** be the RLP encoded block header indicated by the `ephemeral_headers_key.block_hash` field from the content key.
+* The first element in the list **MUST** be the RLP encoded block header indicated by the `find_content_ephemeral_headers_key.block_hash` field from the content key.
 * Each element after the first element in the list **MUST** be the RLP encoded block header indicated by the `header.parent_hash` of the previous item from the list.
-* The list **SHOULD** contain no more than `ephemeral_headers_key.ancestor_count` items.
+* The list **SHOULD** contain no more than `find_content_ephemeral_headers_key.ancestor_count + 1` items.
+
+##### Ephemeral Header by `Offer`
+
+```python
+# Content and content key
+
+offer_ephemeral_header_key = Container(block_hash: Bytes32)
+selector                 = 0x05
+
+OfferEphemeralHeader = Container(
+  header: ByteList[MAX_HEADER_LENGTH], # RLP encoded header in SSZ ByteList
+)
+offer_ephemeral_header = OfferEphemeralHeader(header: rlp.encode(header))
+
+content          = SSZ.serialize(offer_ephemeral_header)
+content_key      = selector + SSZ.serialize(offer_ephemeral_header_key)
+```
+
+* This content type **MUST** only be used for `Offer` requests
+* Headers offered to the client **MUST** be in a strictly consecutive ascending order by `block_number`
+* When offered ephemeral headers, clients should scan the list for a `block_hash` anchored via the external oracle. All headers preceding the anchored header in the list **MUST** be treated as its direct ancestors in order of decreasing height.
```

Comment:
The sentence that I think is missing here is something like:

> An Offer message containing ephemeral headers SHOULD contain multiple offers.  This list of offers MUST...

Reply:
I think it is a little confusing `Offer messages.. contain multiple offers`. 

So an Offer message would contain content keys for `depth of re-org`(this would normally just be the next block so 1) + (I was talking with Milos and he says he wants some constant of additional headers just in case)

So `An "Offer" message SHOULD contain content keys for the "depth of headers since the last lightclientupdate" plus "8 more headers"` Something like this me and Milos were talking about on a call

Reply:
^ This is just the rough idea. But I would like to try and avoid recursively using the word offer

Reply:
Then we were talking about specifying how clients getting this offer should handle re-gossiping this stuff. Which I will formalize today